calc/src/mechanics/util.ts

```diff
@@ -109,11 +109,13 @@ export function getFinalSpeed(gen: Generation, pokemon: Pokemon, field: Field, s
   (pokemon.hasAbility('Chlorophyll') && weather.includes('Sun')) ||
   (pokemon.hasAbility('Swift Swim') && weather.includes('Rain')) ||
   (pokemon.hasAbility('Slush Rush') && ['Hail', 'Snow'].includes(weather)) ||
-  (pokemon.hasAbility('Surge Surfer') && (terrain === 'Electric') || pokemon.hasAbility('Levitate', 'DragonFly')) ||
-  (pokemon.hasAbility('Sand Rush') && weather === 'Sand') ) {
+  (pokemon.hasAbility('Surge Surfer') && (terrain === 'Electric')) ||
+  (pokemon.hasAbility('Sand Rush') && weather === 'Sand')) {
     speedMods.push(6144);
   } else if (pokemon.hasAbilityActive('Slow Start')) {
     speedMods.push(2048);
+  } else if (pokemon.hasAbility('Lead Coat')) {
```

Comment:
i've added this change in the UI, so the pokemon is already slowed down, i've done the same for violent rush.
after a bit of thinking i'll remove what i did in the UI to put it in here.
So yeah if violent rush is active you need to add 6144 to the speedmods too

Reply:
thanks for notifying it to me, it was an UI problem, the abilities active box refreshing automatically the calc is fixed now

Reply:
if you could add the surge surfer that triggers on levitate mechanics, i'll merge

Reply:
I believe this current code works fine? Don't need extra checks since it'll always work whether the target is grounded or not, so as long as the pokemon has surge surfer and electric terrain is up they'll get the speed boost

Reply:
"Surge Surfer	1.5x Speed on Electric Terrain. Works while levitating."

Reply:
my bad i probably missunderstood the move